### PR TITLE
chore: switch to podman setup.exe

### LINKED ISSUE #240: Switch to using podman setup.exe installer as a default podman installation method
Right now we are using zip archive to install podman for the windows workflows. This works quite ok for wsl provider but for hyperv virtualization provider we are encountering problem with `gvproxy` as seen in #228. 

Right now the podman zip url parameter from the workflows:
```
      podman_remote_url:
        default: 'https://github.com/containers/podman/releases/download/v5.2.0/podman-remote-release-windows_amd64.zip'
        description: 'podman remote zip'
        type: string
        required: true
```

Can use zip url to installs podman on path. With use of `pde2e-podman:v0.0.2-windows` image version `setup.exe` is also supported. Installation of the podman binarines via the installer is the only way to use hyperv as podman machine virtualization provider.

### PR DESCRIPTION
**Pull request by danivilla9**
Switches the previous installation of podman (zip archive) to using the setup.exe version. Fixes #240

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
question: how do you determine the version of podman to use ?

podman desktop is using the v5.2.5
https://github.com/containers/podman-desktop/blob/main/extensions/podman/packages/extension/src/podman5.json

**Comment by danivilla9** [comment]
> question: how do you define the version of podman to use ? here I see 5.2.4, before 5.0.3 so it seems for both not in sync with either the latest from podman or the one podman desktop is using
> 
> podman desktop is using the v5.2.5 https://github.com/containers/podman-desktop/blob/main/extensions/podman/packages/extension/src/podman5.json

I used the previous existing version used on other yaml files, I did not realize that it wasn't the most recent one. The older version that was being used earlier (5.0.3) probably was forgotten to be updated as time went on, unless I'm mistaken.

I've updated all instances to v5.2.5 for consistency

**Comment by odockal** [comment]
Regarding inputs. This does not really matters, since the parameter is only used when run on event `workflow_dispatch`. What we really needs it to use particular podman version based on the context of the job.

In [this job](https://github.com/containers/podman-desktop-e2e/blob/main/.github/workflows/podman-desktop-e2e-windows.yaml) we are installing the podman from podman-desktop, so no need to specify any particular podman version.

In [job](https://github.com/containers/podman-desktop-e2e/blob/main/.github/workflows/podman-desktop-e2e-nightly-windows-wsl.yaml) where we test podman-nightly installed from the nightly CI builds we might just set a new version in input parameter, but when run on nightly base on schedule, the latest podman remote version is used.

Where we really need to consider using latest podman that is available or defined by podman-desktop is [this one](https://github.com/containers/podman-desktop-e2e/blob/main/.github/workflows/podman-desktop-e2e-nightly-windows-hyperv.yaml) where we want to use setup.exe podman version as defined in podman-desktop because there is no setup.exe from the podman nightly CI builds, so we need to add a code to define version like in [bootc](https://github.com/containers/podman-desktop-extension-bootc/blob/main/.github/workflows/bootc-e2e-nightly-windows.yaml#L95) and use in when job is triggered  on schedule. Again we can use some particular latest version in input but that does not really matter (although is nice to have latest there).

On the missing two jobs, we just update input default value. that is fine since the job are only run as `workflow_dispatch`. thus only using user's set params.

File: .github/workflows/podman-desktop-e2e-nightly-windows-hyperv.yaml [hunk 1]
```diff
@@ -62,21 +62,29 @@ jobs:
...
+        DEFAULT_VERSION: "${{ env.PD_PODMAN_VERSION || '5.2.5' }}"
+        DEFAULT_URL: 'https://github.com/containers/podman/releases/download/v$DEFAULT_VERSION/podman-$DEFAULT_VERSION-setup.exe'
         DEFAULT_IMAGES_VERSIONS: 'BUILDER="v0.0.2",PODMAN="v0.0.2",RUNNER="v0.0.2"'
       run: |
         echo "FORK=${{ github.event.inputs.fork || env.DEFAULT_FORK }}" >> $GITHUB_ENV
         echo "BRANCH=${{ github.event.inputs.branch || env.DEFAULT_BRANCH }}" >> $GITHUB_ENV
         echo "PODMAN_PROVIDER=${{ github.event.inputs.podman_provider || env.DEFAULT_PODMAN_PROVIDER }}" >> $GITHUB_ENV
         echo "NPM_TARGET=${{ github.event.inputs.npm_target || env.DEFAULT_NPM_TARGET }}" >> $GITHUB_ENV
         echo "ENV_VARS=${{ github.event.inputs.env_vars || env.DEFAULT_ENV_VARS }}" >> $GITHUB_ENV
+        echo "PODMAN_VERSION=${{ github.event.inputs.podman_version || env.DEFAULT_VERSION }}" >> $GITHUB_ENV
```
Comment (odockal) [inline .github/workflows/podman-desktop-e2e-nightly-windows-hyperv.yaml:87]: the version env. var. is not being used in the podman image. please remove.

File: .github/workflows/podman-desktop-e2e-nightly-windows-hyperv.yaml [hunk 2]
```diff
@@ -161,6 +169,7 @@ jobs:
           quay.io/odockal/pde2e-podman:${{ env.PDE2E_PODMAN }}-windows  \
             pd-e2e/podman.ps1 \
               -downloadUrl ${{ env.PODMAN_URL }} \
+              -version ${{ env.PODMAN_VERSION }} \
```
Comment (odockal) [inline .github/workflows/podman-desktop-e2e-nightly-windows-hyperv.yaml:172]: version param not used anymore. please remove.

**Review by odockal — CHANGES_REQUESTED** [review]
Good changes, except version param that is not used anymore. the remote url is the important one.

**Review by odockal — APPROVED** [review]

**Pull request merged**